pkgs/tools/security/yubihsm-shell/default.nix

```diff
@@ -48,12 +48,16 @@ stdenv.mkDerivation rec {
     curl
     openssl
   ] ++ lib.optionals stdenv.isLinux [
-    pcsclite
+    pcsclite.dev
   ] ++ lib.optionals stdenv.isDarwin [
     darwin.apple_sdk.frameworks.PCSC
     libiconv
   ];
 
+  preBuild = lib.optional stdenv.isLinux ''
+    NIX_CFLAGS_COMPILE="$(pkg-config --cflags libpcsclite) $NIX_CFLAGS_COMPILE"
+  '';
```

Comment:
Is `preBuild` meant to be a string or a list?

If it's a list, I would prefer to use `lib.optionals`.
If it's a string, then `lib.optionalString` should be use.

Reply:
changed to `lib.optionalString`